code/model/modelinterp.cpp

```diff
@@ -3603,8 +3603,22 @@ void submodel_get_two_random_points(int model_num, int submodel_num, vec3d *v1,
 		return;
 	}
 
-	int vn1 = (myrand()>>5) % nv;
-	int vn2 = (myrand()>>5) % nv;
+#ifndef NDEBUG
+	if (RAND_MAX < nv)
+	{
+		static int submodel_get_two_random_points_warned = false;
+		if (!submodel_get_two_random_points_warned)
+		{
+			polymodel *pm = model_get(model_num);
+			Warning(LOCATION, "RAND_MAX is only %d, but submodel %d for model %s has %d vertices!  Explosions will not propagate through the entire model!", RAND_MAX, submodel_num, pm->filename, nv);
```

Comment:
The string for `Warning()` is supposed to end with a `\n`, as noted on [the wiki](http://hard-light.net/wiki/index.php/FSO_Error_Handling).